### PR TITLE
add new hardware for retro-go

### PR DESCRIPTION
**Pull request by FantasyGmm**
I don't have other hardware, so I don't know whether other hardware can be compiled and used normally. I try my best to modify it

### CODE REVIEW

File: components/retro-go/netplay/rg_netplay.c
Comment (FantasyGmm) [inline components/retro-go/netplay/rg_netplay.c:350]: I don't remember why I changed this line, maybe it was an accident

Comment (ducalex) [inline components/retro-go/netplay/rg_netplay.c:350]: This should be undone, I think you didn't pull changes I've made before pushing.

File: components/retro-go/rg_display.c
Comment (ducalex) [inline components/retro-go/rg_display.c:160]: This should be undone, I think you didn't pull changes I've made before pushing.

File: components/retro-go/rg_input.c
Comment (ducalex) [inline components/retro-go/rg_input.c:18]: This should be undone, I think you didn't pull changes I've made before pushing.

File: components/retro-go/rg_system.c
Comment (ducalex) [inline components/retro-go/rg_system.c:384]: This should be undone, I think you didn't pull changes I've made before pushing.

File: components/retro-go/rg_system.h
Comment (ducalex) [inline components/retro-go/rg_system.h:32]: This should be undone, I think you didn't pull changes I've made before pushing.

File: components/retro-go/targets/odroid-go.h
Comment (ducalex) [inline components/retro-go/targets/odroid-go.h:13]: I think this whole file should be undone, it doesn't belong in that PR.

File: launcher/main/images.c
Comment (ducalex) [inline launcher/main/images.c:1]: Same here, this whole file should be undone. It doesn't belong in that PR.

File: launcher/main/main.c
Comment (ducalex) [inline launcher/main/main.c:166]: This whole file should also be undone, the reason for these #ifdefs are backwards compatibility, they aren't needed on new devices

File: components/retro-go/rg_display.c
Comment (ducalex) [inline components/retro-go/rg_display.c:143]: The esp32 can also run at 80mhz, the reason it's set to 40mhz is because the display can't support 80mhz.
Are you sure it never glitches on your display?

File: README.md
Comment (ducalex) [inline README.md:36]: On the odroid-go this is definitely button B on both the multi-firmware and the original firmware (It's controlled by CONFIG_BOOTLOADER_NUM_PIN_FACTORY_RESET over there).

**Review by ducalex — CHANGES_REQUESTED** [review]
There are many changes that do not belong in this feaure PR, can you review these?

File: components/retro-go/rg_system.c
Comment (ducalex) [inline components/retro-go/rg_system.c:378]: I think this should be undone, I've modified recovery mode in a way that should be satisfactory to both of us in ea10aad8ef1cf25902ff68a363ab3704a0b8511a.

File: rg_tool.py
Comment (ducalex) [inline rg_tool.py:20]: This changes conflicts with a change I've just made, line should be:

`TARGETS = ["odroid-go"] # We just need to specify the default, the others are discovered below`

Sorry about that!

**Review by ducalex — COMMENTED** [review]
Also please revert rg_images.c, because your PR will delete it.

I think it will be ready to merge after these changes, Thanks!

**Comment by FantasyGmm** [comment]
There are too many changes, I'll delete it and pull up the latest one and then I'm submitting the PR

**Comment by FantasyGmm** [comment]
> Also please revert rg_images.c, because your PR will delete it.
> 
> I think it will be ready to merge after these changes, Thanks!


Unfortunately, I can't roll back. I have to upload it again

**Comment by ducalex** [comment]
> Unfortunately, I can't roll back. I have to upload it again

Yeah I can't merge until it is gone from the Files Changed. 

If you want you can start over with a new PR, I will not push any changes to master for a few days to avoid conflicting with your work.

**Comment by FantasyGmm** [comment]
> > Unfortunately, I can't roll back. I have to upload it again
> 
> Yeah I can't merge until it is gone from the Files Changed.
> 
> If you want you can start over with a new PR. I will avoid pushing changes to master for a few days to not conflict with your work either way.

Unfortunately, my PD power supply has damaged my  board, and I am also waiting for a new PCB.this may delay a little. The old PCB has too many design problems. I redesigned it and changed the pin assignment, so I need to change the code, but I can't test my code now

**Comment by FantasyGmm** [comment]
I will delete the warehouse and fork a new version. After modification, the PCB is available. I will test it

**Pull request closed without merging**
